Not working in gradle because of ${revision}

Working fine in maven project but getting below error for gradle project while building.

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Could not resolve all files for configuration ':compileClasspath'.
   > Could not resolve org.semver4j:semver4j:2.0.1.
     Required by:
         project :
      > Could not resolve org.semver4j:semver4j:2.0.1.
         > inconsistent module metadata found. Descriptor: org.semver4j:semver4j:${revision} Errors: bad version: expected='2.0.1' found='${revision}'
      > Could not resolve org.semver4j:semver4j:2.0.1.
         > inconsistent module metadata found. Descriptor: org.semver4j:semver4j:${revision} Errors: bad version: expected='2.0.1' found='${revision}'
```

**Following change may resolve this error, tested in my local.**
- In pom.xml change ${revision} to particular version i.e 2.0.1

ex : 
```
    <groupId>org.semver4j</groupId>
    <artifactId>semver4j</artifactId>
    <version>2.0.1</version>
    <packaging>jar</packaging>
```

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -18,10 +18,7 @@ jobs:
             const version = tagName.replace('v', '')
             core.setOutput('version', version)
 
-      - name: Build with Apache Maven
-        run: mvn --batch-mode package
-
-      - name: Set up Apache Maven Central
+      - name: Set up Java
         uses: actions/setup-java@v3
         with:
           java-version: 8
@@ -32,8 +29,14 @@ jobs:
           gpg-private-key: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }}
           gpg-passphrase: MAVEN_GPG_PASSPHRASE
 
+      - name: Build with Apache Maven
+        run: mvn --batch-mode package
+
+      - name: Set artifact version
+        run: mvn --batch-mode versions:set -DnewVersion=${{ steps.set_version.outputs.version }}
+
       - name: Publish package to Apache Maven Central
-        run: mvn --batch-mode deploy -Drevision=${{ steps.set_version.outputs.version }}
+        run: mvn --batch-mode deploy
         env:
           MAVEN_USERNAME: ${{ secrets.OSSRH_USERNAME }}
           MAVEN_PASSWORD: ${{ secrets.OSSRH_TOKEN }}

--- a/pom.xml
+++ b/pom.xml
@@ -4,7 +4,7 @@
 
     <groupId>org.semver4j</groupId>
     <artifactId>semver4j</artifactId>
-    <version>${revision}</version>
+    <version>0.0.1-SNAPSHOT</version>
     <packaging>jar</packaging>
 
     <name>semver4j</name>

